chain/src/main/scala/org/bitcoins/chain/blockchain/ChainHandler.scala

```diff
@@ -442,10 +442,21 @@ case class ChainHandler(
       case None => FutureUtil.none
       case Some(blockHeight) =>
         for {
-          tipHash <- getBestBlockHash()
-          tipHeightOpt <- getBlockHeight(tipHash)
+          tips <- blockHeaderDAO.chainTips
+          getNAncestorsFs = tips.map { tip =>
+            blockHeaderDAO.getNAncestors(tip.hashBE, tip.height - blockHeight)
+          }
+          ancestorChains <- Future.sequence(getNAncestorsFs)
         } yield {
-          tipHeightOpt.map(tipHeight => tipHeight - blockHeight + 1)
+          val confs = ancestorChains.flatMap { chain =>
+            if (chain.last.hashBE == blockHash) {
+              Some(chain.head.height - blockHeight + 1)
+            } else None
+          }
+
+          if (confs.nonEmpty) {
+            Some(confs.max)
```

Comment:
Is this right? Don't we want to use the "heaviest" confirmation rather than the highest number?

Reply:
We are using `blockHeaderDAO.chainTips` which means all the tips will have equal work. More than likely everything in `confs` will be equal, but we have to pick one, so for some reason if a block is in a chain equal work chain but longer, then I think that's the better chain to pick.